### PR TITLE
builder changes/update hotshot

### PR DESCRIPTION
**Pull request by move47**
- Introduces two new env variables and removes some unused env variables. 
- Updates to hotshot 0.5.41 and other relevant dependencies.

Edit: Fulfilled. ~~REQUEST TO REVIEWERS: I am not aware of our env variables and where we run `permission less-builder` binary with env variables, I request reviewer to make push those relevant changes before we merge this. CI MIGHT FAIL BECAUSE OF THIS.~~

### CODE REVIEW

File: builder/src/bin/permissionless-builder.rs [hunk 1]
```diff
@@ -110,6 +86,25 @@ pub struct NonPermissionedBuilderOptions {
     /// BUILDER CHANNEL CAPACITY
     #[clap(short, long, env = "ESPRESSO_BUILDER_CHANNEL_CAPACITY")]
     pub channel_capacity: NonZeroUsize,
+
+    /// The amount of time a builder can wait before timing out a request to the API.
+    #[clap(
+        short,
+        long,
+        env = "WEBSERVER_RESPONSE_TIMEOUT_DURATION",
```
Comment (sveitser) [inline builder/src/bin/permissionless-builder.rs:94]: Please prefix with `ESPRESSO_BUILDER`

**Comment by sveitser** [comment]
To answer the questions the env vars are set in the `.env` file and if you want to pass it to docker containers also in `docker-compose.yaml` rarely also in the `process-compose.yaml` file because for the native demo the env vars in `.env` are all available to the binaries.

Here it looks like you have workable default values then you don't need to add anything to any other files.

File: builder/src/bin/permissioned-builder.rs [hunk 2]
```diff
@@ -133,24 +133,43 @@ pub struct PermissionedBuilderOptions {
     pub state_peers: Vec<Url>,
 
     /// Port to run the builder server on.
-    #[clap(short, long, env = "BUILDER_SERVER_PORT")]
+    #[clap(short, long, env = "ESPRESSO_BUILDER_SERVER_PORT")]
     pub port: u16,
 
     /// Port to run the builder server on.
-    #[clap(short, long, env = "BUILDER_ADDRESS")]
+    #[clap(short, long, env = "ESPRESSO_BUILDER_ADDRESS")]
```
Comment (sveitser) [inline builder/src/bin/permissioned-builder.rs:140]: Is this variable used? The address is obtained from the private key derived from the mnemonic.

File: builder/src/bin/permissionless-builder.rs [hunk 3]
```diff
@@ -110,6 +86,25 @@ pub struct NonPermissionedBuilderOptions {
     /// BUILDER CHANNEL CAPACITY
     #[clap(short, long, env = "ESPRESSO_BUILDER_CHANNEL_CAPACITY")]
     pub channel_capacity: NonZeroUsize,
+
+    /// The amount of time a builder can wait before timing out a request to the API.
+    #[clap(
+        short,
+        long,
+        env = "ESPRESSO_WEBSERVER_RESPONSE_TIMEOUT_DURATION",
```
Comment (jbearer) [inline builder/src/bin/permissionless-builder.rs:94]:
```suggestion
        env = "ESPRESSO_BUILDER_WEBSERVER_RESPONSE_TIMEOUT_DURATION",
```

File: builder/src/bin/permissioned-builder.rs (hunk 2 above)
Comment (sveitser) [reply builder/src/bin/permissioned-builder.rs:140]: Actually nevermind, I thought we will remote the permissioned builder anyway, so no need to fix the code I think.

**Review by jbearer — APPROVED** [review]

**Pull request merged**